pkg/newlog/cmd/vector.go

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) 2025 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"net"
+	"os"
+	"slices"
+	"strings"
+	"time"
+)
+
+var (
+	uploadSockVectorSource = "/run/devUpload_source.sock"
+	keepSockVectorSource   = "/run/devKeep_source.sock"
+	uploadSockVectorSink   = "/run/devUpload_sink.sock"
+	keepSockVectorSink     = "/run/devKeep_sink.sock"
+)
+
+type BufferedSockWriter struct {
+	path      string
+	buffer    chan []byte
+	reconnect time.Duration
+}
+
+// NewBufferedSockWriter creates a new buffered socket writer that writes to the specified path.
+// It uses a buffered channel to queue messages and attempts to reconnect if the connection is lost.
+func NewBufferedSockWriter(path string, bufSize int, reconnect time.Duration) *BufferedSockWriter {
+	sw := &BufferedSockWriter{
+		path:      path,
+		buffer:    make(chan []byte, bufSize),
+		reconnect: reconnect,
+	}
+	go sw.run()
+	return sw
+}
+
+func (sw *BufferedSockWriter) run() {
+	for {
+		conn, err := net.Dial("unix", sw.path)
```

Comment:
Just a minor comment here: without a timeout it can hang for minutes if the path exists but nobody’s listening. Not sure it's a realistic scenario in our case.

Reply:
After some consideration I think waiting indefinitely until Vector comes up is the desired behavior. The logging pipeline is not complete without it. At most newlogd can fatal out and cause a device reboot with the hope that Vector will start on the second try